unblob/processing.py

```diff
@@ -136,6 +166,13 @@ def process_file(
         )
         return ProcessResult()
 
+    try:
+        if not hasattr(process_file, "_sandboxed"):
+            sandbox(extract_dir, report_file)
+            process_file._sandboxed = True  # noqa: SLF001
+    except SandboxError:
+        logger.warning("Sandboxing FS access is unavailable on this system, skipping.")
+
```

Comment:
Is it a problem to call this multiple times? If so I'd separate concerns a bit clearer:

```python
import functools

def call_once(fn):
    fn.__called_once__ = False
    
    @functools.wraps(fn)
    def wrapper(*args, **kwargs):
        if fn.__called_once__:
            return
        fn(*args, **kwargs)
        fn.__called_once__ = True

@call_once
def sandbox(...):
    ...
```

This way, we can even test the once-call functionality independently.